source/extensions/filters/http/grpc_field_extraction/message_converter/stream_message.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+#include "envoy/buffer/buffer.h"
+
+#include "source/common/common/logger.h"
+
+#include "proto_field_extraction/message_data/message_data.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace GrpcFieldExtraction {
+
+// A stream message object wraps an individual gRPC message for a gRPC stream.
+// It is neither copyable nor movable.
+class StreamMessage {
+public:
+  StreamMessage(std::unique_ptr<Protobuf::field_extraction::MessageData> message,
+                std::unique_ptr<Envoy::Buffer::Instance> owned_bytes,
+                std::shared_ptr<uint64_t> ref_to_bytes_usage)
+      : message_(std::move(message)), owned_bytes_(std::move(owned_bytes)),
+        ref_to_bytes_usage_(ref_to_bytes_usage) {
+    if (owned_bytes_ != nullptr) {
+      ENVOY_LOG_MISC(info, "owned len(owned_bytes_)={}", owned_bytes_->length());
+      *ref_to_bytes_usage_ += owned_bytes_->length();
+    }
+  }
+
+  StreamMessage(const StreamMessage&) = delete;
+
+  StreamMessage& operator=(const StreamMessage&) = delete;
+
+  ~StreamMessage() {
+    if (owned_bytes_ != nullptr) {
+      ENVOY_LOG_MISC(info, "free len(owned_bytes_)={}", owned_bytes_->length());
```

Comment:
These messages seem a bit verbose. Should they perhaps be trace or debug?

Reply:
Done.